### ISSUE TITLE
JDBI batch insert seems not to be able to bind enum arguments

### ISSUE THREAD

**Issue by aitrusgit** [jdbi/jdbi#1623]
We just updated from this versions:

```
org.jdbi:jdbi3-core:3.10.1
org.jdbi:jdbi3-kotlin-sqlobject:3.10.1
org.jdbi:jdbi3-kotlin:3.10.1
org.jdbi:jdbi3-sqlobject:3.10.1
```

to this:

```
org.jdbi:jdbi3-core:3.11.0
org.jdbi:jdbi3-kotlin-sqlobject:3.11.0
org.jdbi:jdbi3-kotlin:3.11.0
org.jdbi:jdbi3-sqlobject:3.11.0
```

And we start getting an `IllegalStateException` while trying to batch insert beans that contain a property that is an enum type.

An small example that reproduce the issue is the following:

```
/**
 * The test table was created like this:
 *
 *
 *  CREATE TABLE person(
 *      id serial primary key,
 *      name text not null,
 *      age int not null,
 *      gender text not null
 *  )
 *
 */
class TestJDBI {

    @Test
    fun `insert works OK`() {
        val person = Person("Luke", 19, Gender.MALE)
        jdbi.onDemand<Queries>().insertPerson(person)
    }

    @Test
    fun `batch insert doesn't work OK`() {
        val person0 = Person("Luke", 19, Gender.MALE)
        val person1 = Person("Leia", 19, Gender.FEMALE)
        val person2 = Person("C3P0", 45, Gender.OTHER)

        jdbi.onDemand<Queries>().insertPersons(listOf(person0, person1, person2)) //ends up throwing an IllegalStateException
    }

    interface Queries {

        @SqlUpdate("""
            insert into person(name, age, gender)
            values (:name, :age, :gender)
        """)
        fun insertPerson(@BindBean value: Person)

        @SqlBatch("""
            insert into person(name, age, gender)
            values (:name, :age, :gender)
        """)
        fun insertPersons(@BindBean value: List<Person>)
    }

    data class Person(
        val name: String,
        val age: Int,
        val gender: Gender
    )

    enum class Gender {
        MALE,
        FEMALE,
        OTHER
    }
}
```

The non-batch insert works ok while the batch one (which used to work fine in 3.10.1) now throws an `IllegalStateException`

```
java.lang.IllegalStateException
	at org.jdbi.v3.core.statement.PreparedBatch.lambda$buildArgument$1(PreparedBatch.java:87)
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at org.jdbi.v3.core.statement.PreparedBatch.buildArgument(PreparedBatch.java:87)
	at org.jdbi.v3.core.argument.internal.NamedArgumentFinderFactory$Pojo.lambda$null$1(NamedArgumentFinderFactory.java:66)
	at java.base/java.util.Optional.map(Optional.java:265)
	at org.jdbi.v3.core.argument.internal.NamedArgumentFinderFactory$Pojo.lambda$forPojoProps$2(NamedArgumentFinderFactory.java:65)
	at org.jdbi.v3.core.statement.ArgumentBinder$Prepared.lambda$prepareBinder$2(ArgumentBinder.java:205)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.HashMap$KeySpliterator.tryAdvance(HashMap.java:1621)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
	at org.jdbi.v3.core.statement.ArgumentBinder$Prepared.prepareBinder(ArgumentBinder.java:207)
	at org.jdbi.v3.core.statement.ArgumentBinder$Prepared.<init>(ArgumentBinder.java:190)
	at org.jdbi.v3.core.statement.PreparedBatch.internalBatchExecute(PreparedBatch.java:202)
	at org.jdbi.v3.core.statement.PreparedBatch.execute(PreparedBatch.java:102)
	at org.jdbi.v3.core.statement.PreparedBatch.executeAndGetModCount(PreparedBatch.java:110)
	at org.jdbi.v3.sqlobject.statement.internal.SqlBatchHandler.lambda$executeBatch$5(SqlBatchHandler.java:308)
	at org.jdbi.v3.core.transaction.LocalTransactionHandler.inTransaction(LocalTransactionHandler.java:134)
	at org.jdbi.v3.core.Handle.inTransaction(Handle.java:422)
	at org.jdbi.v3.sqlobject.statement.internal.SqlBatchHandler.executeBatch(SqlBatchHandler.java:308)
	at org.jdbi.v3.sqlobject.statement.internal.SqlBatchHandler.access$000(SqlBatchHandler.java:46)
	at org.jdbi.v3.sqlobject.statement.internal.SqlBatchHandler$2.hasNext(SqlBatchHandler.java:206)
	at org.jdbi.v3.sqlobject.statement.internal.SqlBatchHandler.invoke(SqlBatchHandler.java:232)
	at org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.lambda$invoke$0(SqlObjectInitData.java:125)
	at org.jdbi.v3.core.internal.Invocations.invokeWith(Invocations.java:44)
	at org.jdbi.v3.core.internal.Invocations.invokeWith(Invocations.java:26)
	at org.jdbi.v3.core.LazyHandleSupplier.lambda$invokeInContext$1(LazyHandleSupplier.java:77)
	at org.jdbi.v3.core.internal.Invocations.invokeWith(Invocations.java:44)
	at org.jdbi.v3.core.internal.Invocations.invokeWith(Invocations.java:26)
	at org.jdbi.v3.core.LazyHandleSupplier.invokeInContext(LazyHandleSupplier.java:76)
	at org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.call(SqlObjectInitData.java:131)
	at org.jdbi.v3.sqlobject.internal.SqlObjectInitData$1.invoke(SqlObjectInitData.java:125)
	at org.jdbi.v3.sqlobject.SqlObjectFactory.lambda$attach$2(SqlObjectFactory.java:110)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$invoke$5(OnDemandExtensions.java:98)
	at org.jdbi.v3.core.internal.exceptions.Unchecked.lambda$function$4(Unchecked.java:76)
	at org.jdbi.v3.core.internal.OnDemandExtensions.invoke(OnDemandExtensions.java:98)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$null$2(OnDemandExtensions.java:82)
	at org.jdbi.v3.core.Jdbi.callWithExtension(Jdbi.java:475)
	at org.jdbi.v3.core.Jdbi.withExtension(Jdbi.java:462)
	at org.jdbi.v3.core.internal.OnDemandExtensions.lambda$createProxy$3(OnDemandExtensions.java:82)
	at com.sun.proxy.$Proxy122.insertPersons(Unknown Source)
	at com.flourish.position.unit.TestJDBI.batch insert doesn't work OK(TestBatchInsert.kt:39)
```

This seems to be cased by the `:gender` argument.  It works fine if I do hardcode its value in the query like this:

```
       @SqlBatch("""
            insert into person(name, age, gender)
            values (:name, :age, 'OTHER') --:gender
        """)
        fun insertPersons(@BindBean value: List<Person>)
```

**Comment by aitrusgit** [comment]
Seems to be fixed in version `3.11.1`.

**Comment by stevenschlansker** [comment]
Yep, that's exactly the bug fixed in that version.  Thanks for reporting and sorry for the trouble!
